src/com/aca/Manager.java

```diff
@@ -0,0 +1,142 @@
+        this.gadgets = gadgets;
+    }
+
+    public void sellItem(String name) throws IOException {
+        for (Gadget gadget : gadgets) {
+            if (gadget.getName().equals(name)){
+                gadget.setCount(gadget.getCount()-1);
+                soldAmount += gadget.getPrice();
+                Gadget.allItemsCount-=1;
+                int memory = 0;
+                if (gadget instanceof MemoryGadget){
+                    memory = ((MemoryGadget) gadget).getMemoryInGB();
+                }
+                String[] row = new String[]{
+                        gadget.getName(),
+                        String.valueOf(gadget.getDuration()),
+                        String.valueOf(gadget.getPrice()),
+                        String.valueOf(gadget.getCount()),
+                        String.valueOf(memory)
+                };
+
+                FileWriter csvWriter = new FileWriter("aca-java-june-homeworks/src/com/aca/productList/trading.csv");
+
+                    csvWriter.append(String.join(",", row));
+                    csvWriter.append("\n");
+
+
+                csvWriter.flush();
+                csvWriter.close();
+            }
+        }
+    }
+
+    public List<VideoRecorder> getAllVideoRecorders(){
+        List<VideoRecorder> recorders = new ArrayList<>();
+        for (Gadget gadget : gadgets) {
+            if (gadget instanceof VideoRecorder){
+                recorders.add((VideoRecorder)gadget);
+            }
+        }
+        return recorders;
+    }
+
+
+    public List<AudioRecorder> getAllAudioRecorders(){
+        List<AudioRecorder> recorders = new ArrayList<>();
+        for (Gadget gadget : gadgets) {
+            if (gadget instanceof AudioRecorder){
+                recorders.add((AudioRecorder)gadget);
+            }
+        }
+        return recorders;
+    }
+
+
+
+    public List<VideoRecorder> getAllVideoRecordersByType(VideoRecorder.VideoQuality videoQuality){
+        List<VideoRecorder> recorders = new ArrayList<>();
+        for (Gadget gadget : gadgets) {
+            if (gadget instanceof VideoRecorder){
+                VideoRecorder videoRecorder = (VideoRecorder) gadget;
+                if(videoRecorder.getVideoQuality() == videoQuality){
+                    recorders.add(videoRecorder);
+                }
+            }
+        }
+        return recorders;
+    }
+    public List<AudioRecorder> getAllAudioRecordersByType(AudioRecorder.AudioQuality audioQuality){
+        List<AudioRecorder> recorders = new ArrayList<>();
+        for (Gadget gadget : gadgets) {
+            if (gadget instanceof AudioRecorder){
+                AudioRecorder audioRecorder = (AudioRecorder) gadget;
+                if(audioRecorder.getAudioQuality() == audioQuality){
+                    recorders.add(audioRecorder);
+                }
+            }
+        }
+        return recorders;
+    }
+
+    public void getListOfItems(List<Gadget> gadgets){
```

Comment:
Hamo jan getListOfItems is the name , but it mostly prints detils and I don't understand clearly the meaning of this method, which new functionality provides this method or which point covers from requirements